micropython/bluetooth/aioble/aioble/central.py

```diff
@@ -187,10 +188,11 @@ def _decode_field(self, *adv_type):
                     yield payload[i + 2 : i + payload[i] + 1]
                 i += 1 + payload[i]
 
-    # Returns the value of the advertised name, otherwise empty string.
+    # Returns the value of the complete (or shortened) advertised name, otherwise empty string.
     def name(self):
-        for n in self._decode_field(_ADV_TYPE_NAME):
+        for n in self._decode_field(_ADV_TYPE_NAME, _ADV_TYPE_SHORT_NAME):
             return str(n, "utf-8") if n else ""
+        return ""
```

Comment:
Hrmm. I realise this is now matching the comment. But I think this should actually be the old behavior (the comment was incorrect). i.e. Returning None means there was no name field, as opposed to empty string which would be the (somewhat odd) case of there being an empty name set.

Sorry! Could you please fix the comment instead.

Reply:
That's definitely more logical, `None` literally meaning there isn't a name available.  I'm on it.